### PR TITLE
General upgrade of AKS and all support projects

### PR DESCRIPTION
**Pull request by danquah**
#### What does this PR do?
The original goal of this change was to upgrade AKS, but due to some problems with the Terraform provider I ended up rebuilding the entire environment.

As I had to rebuild the entrie cluster, and as we do not have any other environements, I took the chance to upgrade all support projects as well.

#### Should this be tested by the reviewer and how?
You will have to go through all the commits. I've split them up into the individual changes for easier review.

#### What are the relevant tickets?
DDFDPDEL-212

### CODE REVIEW

File: infrastructure/task/scripts/util.source.sh [hunk 1]
```diff
@@ -19,8 +19,10 @@ function setupHelmRepo() {
   local alias=$1
   local url=$2
   # See if we can find the repo in the list of repos, if not add it
-  helm repo list | grep "^${alias}\b" || helm repo add "${alias}" "${url}"
-  helm repo update "${alias}"
+  if ! helm repo list | grep "^${alias}\b"; then
+    helm repo add "${alias}" "${url}"
+    helm repo update "${alias}"
+  fi
```
Comment (achton) [inline infrastructure/task/scripts/util.source.sh:25]: Nice optimization :wink:

File: documentation/runbooks/upgrading-aks.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# Upgrading AKS
```
Comment (achton) [inline documentation/runbooks/upgrading-aks.md:1]: 🙏🏻

File: documentation/runbooks/upgrading-aks.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+## Procedure
+
+We use Terraform to upgrade AKS. Should you need to do a manual upgrade consult
+Azures documentation on [upgrading a cluster](https://learn.microsoft.com/en-us/azure/aks/upgrade-cluster)
+and on [upgrading node pools](https://learn.microsoft.com/en-us/azure/aks/use-multiple-node-pools#upgrade-a-node-pool).
+Be aware in both cases that the Terraform state needs to be brought into sync
+via some means, so this is not a recommended approach.
+
+## 1. find out which versions of kubernetes an environment can upgrade to
```
Comment (achton) [inline documentation/runbooks/upgrading-aks.md:29]: This heading is a bit weirdly formatted.

**Review by achton — APPROVED** [review]
Looks great from where I'm standing! 👍🏻

**Review by achton — CHANGES_REQUESTED** [review]
Oh wait - I forgot I found a couple of things 😂

**Comment by danquah** [comment]
@achton implemented some fixes

**Review by achton — APPROVED** [review]
LGTM! 👍🏻

**Pull request merged**